docs/packaging-your-sql-project/intro-to-etl-pipelines-with-python-and-sql.md

```diff
@@ -11,4 +11,186 @@ kernelspec:
+
+```python
+url_open_canada = "https://open.canada.ca/data/api/action/package_show?id=98f1a129-f628-4ce4-b24d-6f16bf24dd64"  # noqa E501
+json_resp = requests.get(url_open_canada)
+# Check response is successful and application is of type JSON
+if (
+    json_resp.status_code == 200
+    and "application/json" in json_resp.headers.get("Content-Type", "")
+):
+    # Format data and obtain entries in english
+    open_canada_data = json_resp.json()
+    data_entries = pd.json_normalize(
+        open_canada_data["result"], record_path="resources"
+    )
+    data_entries["language"] = data_entries["language"].apply(
+        lambda col: col[0]
+    )
+    data_entries_english = data_entries[
+        data_entries["language"] == "en"
+    ]  # noqa E501
+    final_result = data_entries_english[["name", "url"]]
+```
+
+We then finalize the extraction process by saving each data set in its raw format as its own `pandas` data frame from  `final_result`. This is done with the `extract_raw_data` functions under a `for loop` in the script's main call on lines 342 - 358.
+
+```python
+# Fuel consumption metadata extraction urls
+data_entries_english = fuel_consumption_metadata_extraction()
+
+for item in data_entries_english.iterrows():
+    name, url = item[1]["name"], item[1]["url"]
+
+    if "Original" in name:
+        continue
+
+    # Form file name
+    file_name = f'{name.replace(" ","_")}.csv'
+
+    # Extract raw data
+    item_based_url = extract_raw_data(url)
+
+    # Read and clean as pandas df
+    df = pd.read_csv(StringIO(item_based_url.text), low_memory=False)
+```
+
+
+### Transforming with Python `pandas`
+
+We clean each of the extracted data sets with the `read_and_clean` function prior to any other transformations. This function is called immediately after the above lines on line 359. 
+
+```python
+final_df = read_and_clean_df(df)
+```
+
+A large bulk of the script is dedicated to transforming the data into our desired format. Under the `for loop` and after the extraction process described in the above section, we condition the values of the `name` column of our `data_entries_english` `pandas` data frame that contains the name and url of each data set. The script then perform the appropriate transformations conditioned on whether the data set regards hybrid, electric, or gas type vehicles. 
+
+Much of the `pandas` transformations in this section are repeatedly used throughout this script. These functions are easily inferred based on their syntax, such as `replace`, `rename`, and `map`. Because of how repetitive and lengthy the code is, we won't show it in detail for this section. Rather, the main takeaway is that `pandas` transformations are not that difficult to grasp and that they do not stray too far away from SQL functionality. 
```

Comment:
> Much of the `pandas` transformations in this section are repeatedly used throughout this script. These functions are easily inferred based on their syntax, such as `replace`, `rename`, and `map`. Because of how repetitive and lengthy the code is, we won't show it in detail for this section. Rather, the main takeaway is that `pandas` transformations are not that difficult to grasp and that they do not stray too far away from SQL functionality.

This paragraph doesn't help me as a learner, remove if it can't be explained using simple terms

Reply:
Removed.